pkg/generate/app/cmd/newapp_test.go

```diff
@@ -53,6 +53,17 @@ func TestValidate(t *testing.T) {
 			env:                 map[string]string{"one": "first", "two": "second", "three": "third"},
 			parms:               map[string]string{},
 		},
+		"envfile": {
+			cfg: AppConfig{
+				GenerationInputs: GenerationInputs{
+					EnvironmentFile: "/tmp/envfile1",
+				},
+			},
+			componentValues:     []string{},
+			sourceRepoLocations: []string{},
+			env:                 map[string]string{"one": "first", "two": "second", "three": "third"},
+			parms:               map[string]string{},
+		},
```

Comment:
you trust the loop on line 107.  but i'm not sure what this test is actually going to do, at best i'd expect it to get an error saying it couldn't open /tmp/envfile1.  Which is a valid test (testing the error handling) but not the only necessary test.

Also because of the way this particular test works, it's calling into the newapp code after the argument processing has been done, so it's not actually going to attempt to open the file you've specified.  You need something that's actually going to call validate.  Perhaps even a new test case that explicilty calls validate after setting up an AppConfig object, and then confirms that cfg.Environment reflects the correct values afterwards (or threw the correct error)